packages/sn-controls-react/src/fieldcontrols/Boolean.tsx

```diff
@@ -6,78 +6,47 @@ import FormControl from '@material-ui/core/FormControl'
- */
-export class BooleanComponent extends Component<ReactClientFieldSetting, BooleanState> {
-  /**
-   * constructor
-   * @param {object} props
-   */
+export function BooleanComponent(props: ReactClientFieldSetting<FieldSetting>) {
+  const initialState = props.fieldValue != null ? !!props.fieldValue : !!props.settings.DefaultValue
+  const [value, setValue] = useState(initialState)
 
-  constructor(props: BooleanComponent['props']) {
-    super(props)
-    this.state = {
-      value:
-        this.props.content && this.props.content[this.props.settings.Name] != null
-          ? this.props.content[this.props.settings.Name]
-          : this.props.settings.DefaultValue,
-    }
-    this.handleChange = this.handleChange.bind(this)
-  }
-  /**
-   * set selected value
-   */
-  public handleChange = () => {
-    const { value } = this.state
-    const newValue = !value
-    this.setState({
-      value: newValue,
-    })
-    this.props.fieldOnChange && this.props.fieldOnChange(this.props.settings.Name, newValue)
+  const handleChange = () => {
+    setValue(!value)
+    props.fieldOnChange && props.fieldOnChange(props.settings.Name, !value)
   }
-  /**
-   * render
-   * @return {ReactElement} markup
-   */
-  public render() {
-    switch (this.props.actionName) {
-      case 'edit':
-      case 'new':
-        return (
-          <FormControl required={this.props.settings.Compulsory}>
-            <FormControlLabel
-              control={<Checkbox checked={this.state.value} onChange={this.handleChange} />}
-              label={this.props.settings.DisplayName}
-            />
-            {this.props.settings.Description ? (
-              <FormHelperText>{this.props.settings.Description}</FormHelperText>
-            ) : null}
-          </FormControl>
-        )
-      case 'browse':
-      default:
-        return this.props.content && this.props.content[this.props.settings.Name] != null ? (
-          <FormControl component={'fieldset' as 'div'}>
-            <InputLabel shrink={true} htmlFor={name as string}>
-              {this.props.settings.DisplayName}
-            </InputLabel>
-            {this.props.renderIcon
-              ? this.props.renderIcon(this.state.value ? 'check' : 'not_interested')
-              : renderIconDefault(this.state.value ? 'check' : 'not_interested')}
-          </FormControl>
-        ) : null
-    }
+
+  switch (props.actionName) {
+    case 'edit':
+    case 'new':
+      return (
+        <FormControl required={props.settings.Compulsory} disabled={props.settings.ReadOnly}>
+          <FormControlLabel
+            name={props.settings.Name}
+            control={<Checkbox checked={value} onChange={handleChange} />}
+            label={props.settings.DisplayName}
+          />
+          <FormHelperText>{props.settings.Description}</FormHelperText>
+        </FormControl>
+      )
+    case 'browse':
```

Comment:
Boolean in browse mode is bit messed up
![image](https://user-images.githubusercontent.com/6968860/61532941-b9db8d80-aa2b-11e9-8b7f-8eee62b7b691.png)

Reply:
fixed